### ISSUE TITLE
Map child-ids in OTel attributes

### ISSUE THREAD

**Issue by AlexanderWert** [elastic/apm-data#175]
To support inferred spans in OTel-based agents, we need to map child-ids field from OTel attributes to the `child.id` field.

TBD: the OTel attribute name to map from.

**Comment by JonasKunz** [comment]
Update: We decided to instead send the children as span-links with a special marker attribute on the link.

**Comment by axw** [comment]
Clever :)

Is that a permanent solution, or would we still consider creating an OTEP to extend the Span top-level fields with child IDs?

**Comment by AlexanderWert** [comment]
> Is that a permanent solution, or would we still consider creating an OTEP to extend the Span top-level fields with child IDs?

Realistically, I think that's the best solution for OTel that is compliant with the data model. I doubt child IDs is something we could easily get into the data model.

**Comment by JonasKunz** [comment]
> Realistically, I think that's the best solution for OTel that is compliant with the data model.

I agree. Our `child_id`s are just links to some other spans with a special semantic, namely overriding the parent-child relationship. Using span links here with a special attribute on the link to specify the semantic seems conceptually the most correct to me.

We don't have to change anything on how the data is stored in ES and handled by the UI for now, apm-data will translate those special links to `child_id`s instead. If we want to have the "link with special semantic" concept in our data in ES, we can always add that later I think.
